### ISSUE TITLE
Duplicate (double) documentation generated for friend function

### ISSUE THREAD

**Issue by scott-zhong** [doxygen/doxygen#10828]
**Describe the bug**
Duplicate (double) documentation generated for friend function. See the following testcase

```
/**
 * A docs
 */
class A
{
public:
    friend A operator*(const A& x, const A& y);
};

/**
 * \relates A
 * 
 * A operator* docs
 */
A operator*(const A& x, const A& y);

/**
 * B docs
 */
template <typename T>
class B
{
public:
    /**
     * B operator* docs
     */
    T operator*(void) const
    {
        return T();
    }
};
```

`A operator*()` is documented twice in Doxygen 1.8.18 - Doxygen trunk (eb110639dcc32804b59abdf0dcb343abde211b1e*)

**Screenshots**
Doxygen 1.8.17:
![image](https://github.com/doxygen/doxygen/assets/32584126/88c939dc-096c-429f-b018-b27b7577c3ff)

Doxygen 1.8.18 - trunk (eb110639dcc32804b59abdf0dcb343abde211b1e*)
![image](https://github.com/doxygen/doxygen/assets/32584126/fa3ab51a-4307-4f63-8d22-5c0d6a0a3c14)

**To Reproduce**
[testcase.zip](https://github.com/doxygen/doxygen/files/15123063/testcase.zip)

**Expected behavior**
Single instance of `A operator*()` documentation

**Version**
1.8.18 - trunk (eb110639dcc32804b59abdf0dcb343abde211b1e*)

**Comment by scott-zhong** [comment]
I think this is the same bug with template and friend function.

```
/**
 * A docs
 */
template< class T >
class A
{
public:

    friend A<double> fun(const A<double>& v);
};

/**
 * \relates A
 * fun docs
 */
A<double> fun(const A<double>& V);

/**
 * B docs
 */
template< class T >
class B
{
public:

    friend B<double> fun(const B<double>& v);
};
```

1.8.17
![image](https://github.com/doxygen/doxygen/assets/32584126/aaa6c1bb-94a1-4ecd-8804-fedb9b55b505)

1.8.18 - trunk (https://github.com/doxygen/doxygen/commit/eb110639dcc32804b59abdf0dcb343abde211b1e)
![image](https://github.com/doxygen/doxygen/assets/32584126/4f9d01cd-54a7-496e-ab1a-83e19e80ab05)

**Comment by doxygen** [comment]
@scott-zhong Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by scott-zhong** [comment]
@doxygen This does fix it. Thank you!!!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
